explorer/client/src/utils/stringUtils.ts

```diff
@@ -48,6 +50,47 @@ export function truncate(fullStr: string, strLen: number, separator?: string) {
     );
 }
 
+export async function extractFileType(
+    displayString: string,
+    signal: AbortSignal
+) {
+    let result: string;
+
+    try {
+        // First check Content-Type in header:
+        result = await fetch(transformURL(displayString), {
+            signal: signal,
+        })
+            .then(
+                (resp) =>
+                    resp?.headers
+                        ?.get('Content-Type')
+                        ?.split('/')
+                        .reverse()?.[0]
+                        ?.toUpperCase() || 'Image'
+            )
+            .catch((err) => {
+                console.error(err);
+                return 'Image';
+            });
+
+        // When Content-Type cannot be accessed (e.g. because of CORS), rely on file extension
+
+        if (result === 'Image') {
+            const extension =
+                displayString?.split('.').reverse()?.[0]?.toUpperCase() || '';
+            if (['JPG', 'JPEG', 'PNG'].includes(extension)) {
+                result = extension;
+            }
+        }
+    } catch (err) {
+        console.error(err);
+        result = 'Image';
+    }
+
+    return `1 ${result} File`;
```

Comment:
the function is supposed to `extractFileType` but it seems to combine the job of extracting the file type with generating the message for the user.  could we have this function just return the file type ?

Reply:
The message for the user is the file type. Currently, the website only displays single image files - and so the type is specified as `1 [some kind of image] File`. In the future there may be multiple images being displayed or other types of file being handled.

Reply:
`1 [png, jpeg, etc.] File` is not a file type - things like `png`, `jpeg` are.

Reply:
I've split extracting file type (jpg/png/jpeg) from generating the message `1 JPG/PNG/JPEG File`.